api-model.js

```diff
@@ -21,20 +21,48 @@ exports.fetchArticleById = (article_id) => {
 };
 
 exports.fetchArticles = () => {
-  let sqlQuery = "SELECT articles.article_id, articles.title, articles.author, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC;"
+  let sqlQuery =
+    "SELECT articles.article_id, articles.title, articles.author, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC;";
 
   return db.query(sqlQuery).then(({ rows }) => {
-    return rows
-  })
-}
+    return rows;
+  });
+};
 
 exports.fetchCommentsById = (article_id) => {
-  let sqlQuery = "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;"
+  let sqlQuery =
+    "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;";
 
   return db.query(sqlQuery, [article_id]).then(({ rows }) => {
     if (rows.length === 0) {
-      return Promise.reject({ status: 404, msg: "Not found" })
+      return Promise.reject({ status: 404, msg: "Not found" });
     }
-    return rows
-  })
-}
\ No newline at end of file
+    return rows;
+  });
+};
+
+exports.checkUsernameExists = (username) => {
+  return db
+    .query(`SELECT * FROM users WHERE username = $1`, [username])
+    .then((result) => result.rows.length > 0);
+};
+
+exports.addComment = (comment) => {
+  const { article_id, username, body } = comment;
+
+  if (comment.article_id > 13) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  return this.checkUsernameExists(username).then((usernameExists) => {
```

Comment:
Your checkUsernameExists should be doing the Promise.reject which will be caught by your controller's catch block.